app/controllers/support/remaining_device_counts_controller.rb

```diff
@@ -0,0 +1,21 @@
+class Support::RemainingDeviceCountsController < Support::BaseController
+  before_action { authorize Support::ServicePerformance }
+
+  def index
+    respond_to do |format|
+      format.csv do
+        send_data RemainingDeviceCount.to_csv, filename: "remaining_device_counts-#{file_date}.csv"
+      end
+    end
+  end
+
+private
+
+  def file_date
+    RemainingDeviceCount.most_recent.date_of_count.strftime('%Y%m%d')
```

Comment:
I think `RemainingDeviceCount.most_recent` would be nil on the day you deploy this because there won't be any records until that evening.

Reply:
Good spot!